src/sinks/aws_kinesis/firehose/record.rs

```diff
@@ -46,14 +49,26 @@ impl SendRecord for KinesisFirehoseClient {
     type T = KinesisRecord;
     type E = KinesisError;
 
-    async fn send(&self, records: Vec<Self::T>, stream_name: String) -> Option<SdkError<Self::E>> {
+    async fn send(
+        &self,
+        records: Vec<Self::T>,
+        stream_name: String,
+    ) -> Result<KinesisResponse, SdkError<Self::E>> {
+        let rec_count = records.len();
+        let total_size = records.iter().fold(0, |acc, record| {
+            acc + record.data().unwrap_or(&Blob::new(vec![])).as_ref().len()
```

Comment:
Just wondering how common the scenario would be that the record data would be `None`? 
Noticing that we could avoid the `Blob::new(vec![])` , by doing an `if let Some(data) = record.data()` to compute the size and in the else case just use 0.

Reply:
As an alternative, we can use a map here to avoid any unnecessary allocations.
```suggestion
            acc + record.data().map(|v| v.as_ref().len()).unwrap_or_default()
```

Reply:
Fixed